src/bluefinClient.ts

```diff
@@ -1538,6 +1540,298 @@ export class BluefinClient {
+        );
+
+        if (sponsorTxResponse?.ok) {
+          return {
+            ok: true,
+            code: 200,
+            data: sponsorTxResponse,
+            message: "Closed delisted Positions and Withdraw Margin Successful",
+          };
+        }
+
+        if (
+          !sponsorTxResponse?.ok &&
+          sponsorTxResponse?.message !== USER_REJECTED_MESSAGE
+        ) {
+          return this.closeAllDelistedPositionsWithdrawSwapAndDepositToPro({
+            ...args,
+            sponsor: false,
+          });
+        }
+      }
+
+      //unsponsored call
+      return await this.contractCalls.closeAllPositionsAndWithdrawMarginPTB(
+        delistedUserPositionsSymbols,
+        args
+      );
+    } catch (e) {
+      return {
+        ok: false,
+        code: 400,
+        data: "",
+        message: e.message,
+      };
+    }
+  };
+
+  closeAllDelistedPositionsWithdrawSwapAndDepositToPro = async (args?: {
+    gasBudget?: number;
+    sponsor?: boolean;
+    use7k?: boolean;
+    slippage?: number;
+    dryRunOnly?: boolean;
+    defaultSlippage?: number;
+  }) => {
+    try {
+      //get exchange info of all markets to read the status
+      const exchangeInfo = await this.getExchangeInfo();
+      const delistedMarkets = filterDelistedMarkets(exchangeInfo);
+
+      if (delistedMarkets.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted markets",
+        };
+      }
+
+      //get user positions
+      const userPositions = await this.getUserPosition({});
+
+      let delistedUserPositionsSymbols: string[] = [];
+      //get delisted market positions
+      if (userPositions.data && userPositions.data.length > 0) {
+        delistedUserPositionsSymbols = userPositions.data
+          .filter((position) => {
+            return delistedMarkets.includes(position.symbol);
+          })
+          .map((position) => position.symbol);
+      }
+
+      if (delistedUserPositionsSymbols.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted positions",
+        };
+      }
+
+      // cancel open orders for delisted markets
+      await Promise.all(
+        delistedUserPositionsSymbols.map((symbol) =>
+          this.cancelAllOpenOrders(symbol)
+        )
+      );
+
+      if (args?.sponsor) {
+        const sponsoredPayload =
+          await this.contractCalls.closeAllPositionsdWithdrawSwapAndDepositToProPTB(
+            delistedUserPositionsSymbols,
+            args
+          );
+        const sponsorTxResponse = await this.signAndExecuteSponsoredTx(
+          sponsoredPayload
+        );
+
+        if (sponsorTxResponse?.ok) {
+          return {
+            ok: true,
+            code: 200,
+            data: sponsorTxResponse,
+            message:
+              "Closed delisted Positions, withdrew swap and deposited to Pro Successful",
```

Comment:
Let's update this message to "Positions closed, margin withdrawn, swapped for native USDC, and deposited to Pro successfully"